### ISSUE TITLE
modify() within group_by() in 0.8.2 not working.

### ISSUE THREAD

**Issue by ajkroeg** [tidyverse/dplyr#4451]
Hi there,

The new release of dplyr is no longer allowing me do run modify() on data grouped by group_by().  Never had a problem before.  It doesn't give an error of any kind, it just runs forever.  Thanks!

```r
library(dplyr)
df<-data.frame(
  var1=sample(x=1:10, size=16000000, replace=T),
  groups=sample(x=1:100175,size=16000000, replace=T)
)
df<-df%>%
  group_by(groups)%>%
  mutate(bigmean=ifelse(mean(var1)>5,'yes','no'))%>%
  ungroup()
```

**Comment by IndrajeetPatil** [comment]
Maybe the same issue as https://github.com/tidyverse/dplyr/issues/4449?

**Comment by ajkroeg** [comment]
Yeah, I saw that one as well.  Guessing the two problems are related, but since it involves different functions, I figured I should open a new issue.

**Comment by yutannihilation** [comment]
> It doesn't give an error of any kind, it just runs forever.

I could reproduce this issue (and confirmed it used to work v0.8.1). This might be a performance degradation because the code runs if I reduce the `size`.

**Comment by angusmoore** [comment]
This issue seems to be broader than just `modify`. I'm seeing a significant performance regression (in the order of 1000x) for `filter` code too:
```
library(dplyr)
foo <- data.frame(a = rep(1:50000, length.out = 100000),
                  b = rep(1:10000, length.out = 100000),
                  c = runif(100000))

foo %>% filter(a==1, b==1) # Runs fine
foo %>% group_by(a,b) %>% filter(a==1, b==1) # Takes ~1000x longer
```
This example is obviously pretty contrived, but I think it narrows in on the problem in my actual code as much as I could.

**Comment by romainfrancois** [comment]
I believe this is the same as #4458 and fixed in #4461. 

``` r
library(dplyr)

foo <- data.frame(a = rep(1:50000, length.out = 100000),
  b = rep(1:10000, length.out = 100000),
  c = runif(100000))

foo_g <- foo %>% group_by(a,b)
system.time(foo   %>% filter(a==1, b==1))# Runs fine
#>    user  system elapsed 
#>   0.009   0.000   0.010
system.time(foo_g %>% filter(a==1, b==1)) # Takes ~1000x longer
#>    user  system elapsed 
#>   0.908   0.032   0.941
```

and the original example: 

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
df<-data.frame(
  var1=sample(x=1:10, size=16000000, replace=T),
  groups=sample(x=1:100175,size=16000000, replace=T)
)
system.time(df%>%
  group_by(groups)%>%
  mutate(bigmean=ifelse(mean(var1)>5,'yes','no'))%>%
  ungroup())
#>    user  system elapsed 
#>  17.817   0.337  18.160
```

**Comment by ajkroeg** [comment]
Yep, that fixed on my end as well.  Thanks!
